form-flow-valtimo/src/test/kotlin/com/ritense/valtimo/formflow/web/rest/FormFlowResourceTest.kt

```diff
@@ -0,0 +1,91 @@
+package com.ritense.valtimo.formflow.web.rest
+
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.whenever
+import com.ritense.formflow.domain.definition.FormFlowDefinition
+import com.ritense.formflow.domain.definition.FormFlowDefinitionId
+import com.ritense.formflow.domain.definition.FormFlowStep
+import com.ritense.formflow.domain.definition.FormFlowStepId
+import com.ritense.formflow.domain.definition.configuration.FormFlowStepType
+import com.ritense.formflow.domain.definition.configuration.step.FormStepTypeProperties
+import com.ritense.formflow.domain.instance.FormFlowInstance
+import com.ritense.formflow.domain.instance.FormFlowInstanceId
+import com.ritense.formflow.domain.instance.FormFlowStepInstance
+import com.ritense.formflow.domain.instance.FormFlowStepInstanceId
+import com.ritense.formflow.service.FormFlowService
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.springframework.http.MediaType
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+import org.springframework.test.web.servlet.result.MockMvcResultHandlers.print
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status
+import org.springframework.test.web.servlet.setup.MockMvcBuilders
+
+class FormFlowResourceTest {
+    lateinit var mockMvc: MockMvc
+    lateinit var formFlowResource: FormFlowResource
+    lateinit var formFlowService: FormFlowService
+    lateinit var formFlowInstance: FormFlowInstance
+    lateinit var formFlowInstanceId: FormFlowInstanceId
+
+    @BeforeEach
+    fun setUp() {
+        formFlowService = mock()
+
+        formFlowInstanceId = FormFlowInstanceId.newId()
+        formFlowInstance = mock()
+        whenever(formFlowInstance.id).thenReturn(formFlowInstanceId)
+        whenever(formFlowService.getInstanceById(formFlowInstance.id)).thenReturn(formFlowInstance)
+        formFlowResource = FormFlowResource(formFlowService)
+
+        mockMvc = MockMvcBuilders.standaloneSetup(formFlowResource).build()
+    }
+
+    @Test
+    fun `should return form flow state`() {
```

Comment:
Does this provide enough coverage? Right now all that's tested is what happens when everything goes right, but I think testing failure states might be good too.

Reply:
Will add more test(s)